src/rpc/stun.js

```diff
@@ -617,7 +618,7 @@ function test() {
                 return P.join(
                         P.ninvoke(socket, 'send', req, 0, req.length, stun_url.port, stun_url.hostname),
                         P.ninvoke(socket, 'send', ind, 0, ind.length, stun_url.port, stun_url.hostname))
-                    .delay(stun.INDICATION_INTERVAL * chance.floating(stun.INDICATION_JITTER))
+                    .then(promise_utils.delay(stun.INDICATION_INTERVAL * chance.floating(stun.INDICATION_JITTER)))
```

Comment:
OK this is a pitfall.
You must call the delay only when you actually need to start waiting, which is inside the promise chain, and not from the outside function. So please make sure to always use like this `.then(() => promise_utils.delay(555))`

```suggestion
                    .then(() => promise_utils.delay(stun.INDICATION_INTERVAL * chance.floating(stun.INDICATION_JITTER)))
```

Reply:
BTW notice that for all the cases you use it with await there is no need to use an arrow function because that's only needed for `.then(() => ...)`